### PR TITLE
[1.5] Add `waitUntilContainsElement()` to Page

### PR DESCRIPTION
**Pull request by p4veI**
Hello, Ive been working with the chrome lib for the past week in our project and I was unable to crawl through a js enhanced website forms without the ability to trigger some delay programmatically, unless including a long delay via `connectionDelay` option.

So I've used some parts of resolving the element in find methods and passed it through the timeout utility to wait for the element to appear with `waitUntilContainsElement`, hoping we could merge this into the main repo or would you choose a different approach?

Thx!

### CODE REVIEW

File: src/Page.php [hunk 1]
```diff
@@ -440,6 +440,50 @@ private function waitForReloadGenerator($eventName, $loaderId)
...
+     * Wait until page contains Node.
+     *
+     * @param string $selectors
+     * @param int    $timeout
+     *
+     * @throws Exception\OperationTimedOut
+     *
+     * @return $this
+     */
+    public function waitUntilContainsElement(string $selectors, int $timeout = 30000)
```
Comment (enricodias) [inline src/Page.php]: Can we have a return type here?

File: src/Page.php [hunk 2]
```diff
@@ -440,6 +440,50 @@ private function waitForReloadGenerator($eventName, $loaderId)
...
+     */
+    public function waitForElement(string $selectors, int $position = 1)
+    {
+        $delay = 500;
+        $element = [];
+
+        while (true) {
+            try {
+                $element = Utils::getElementPositionFromPage($this, $selectors, $position);
+            } catch (\Throwable $exception) {
```
Comment (enricodias) [inline src/Page.php]: Why catching `\Throwable` instead of exceptions? This may hide php errors.

File: src/Page.php [hunk 3]
```diff
@@ -440,6 +441,48 @@ private function waitForReloadGenerator($eventName, $loaderId)
...
 
+    /**
+     * Wait until page contains Node.
+     *
+     * @param string $selectors
+     * @param int    $timeout
+     *
+     * @throws Exception\OperationTimedOut
+     */
+    public function waitUntilContainsElement(string $selectors, int $timeout = 30000) : self
```
Comment (enricodias) [inline src/Page.php]:
```suggestion
    public function waitUntilContainsElement(string $selectors, int $timeout = 30000): self
```
To make it pass on the styleci check.

File: src/Page.php (hunk 2 above)
Comment (p4veI) [reply src/Page.php]: Yeah, thx, I originally experienced some other error but I resolved that later and was not relevant to the resolving here anwyays.

**Review by enricodias — APPROVED** [review]

File: src/Page.php [hunk 4]
```diff
@@ -440,6 +441,48 @@ private function waitForReloadGenerator($eventName, $loaderId)
...
+
+        return $this;
+    }
+
+    /**
+     * @return bool|\Generator
+     *
+     * @internal
+     */
+    public function waitForElement(string $selectors, int $position = 1)
```
Comment (momala454) [inline src/Page.php:466]: should probably add a timeout option to prevent infinite loot if the element never appear

File: src/Utils.php [hunk 5]
```diff
@@ -90,4 +91,24 @@ public static function closeAllPage(Connection $connection): void
             }
         }
     }
+
+    /**
+     * @throws CommunicationException
+     * @throws Exception\EvaluationFailed
+     *
+     * @return mixed|null
```
Comment (GrahamCampbell) [inline src/Utils.php]: invalid return type. mixed already contains null. please remove the `|null`

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Utils.php (hunk 5 above)
Comment (p4veI) [reply src/Utils.php]: ah sorry, removed. Inferred from IDE tooltip of getResult 😄

**Pull request merged**